src/coding_theory/hamming.lean

```diff
@@ -0,0 +1,245 @@
+/-
+Copyright (c) 2022 Wrenna Robson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wrenna Robson
+-/
+
+import analysis.normed_space.basic
+
+/-!
+# Hamming spaces
+
+The Hamming metric counts the number of places two members of a (finite) Pi type
+differ. The Hamming norm is the same as the Hamming metric over additive groups, and
+counts the number of places a member of a (finite) Pi type differs from zero.
+
+This is a useful notion in various applications, but in particular it is relevant
+in coding theory, in which it is fundamental for defining the minimum distance of a
+code.
+
+In this file we define `hamm β`, the type synonym of a Pi type with the Hamming
+distance `hamm_dist` and weight `hamm_wt` attached, and the various instances that arise
+from the properties of these definitions.
+
+-/
+
+/-
+We define `hamm_dist` and `hamm_wt` over Pi types, and will later attach them to our type
+synonym.
+-/
```

Comment:
```suggestion
In this file we define `hamm β`, the type synonym of a Pi type with the Hamming
distance `hamm_dist` and weight `hamm_wt` attached, and the various instances that arise
from the properties of these definitions.
-/
```
This second comment feels redudant now. Feel free to combine them in a different way, but I don't think we need two separate versions any more.